api/unstable/npn.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include <s2n.h>
+
+/**
+ * @file npn.h
+ *
+ * The Next Protocol Negotiation Extension, or NPN, was an RFC proposal to
+ * negotiate an application protocol. This proposal was never standardized, and
+ * it was eventually replaced with the ALPN Extension. Unfortunately, an early
+ * draft version of the NPN Extension was implemented in several TLS libraries.
+ * Now, clients and servers using older TLS implementations may require this 
```

Comment:
This is why I was asking "which TLS libraries." Do we actually know that it's supported by anything other than Openssl?
```suggestion
 * draft version of the NPN Extension was implemented in OpenSSL.
 * Now, OpenSSL clients and servers may require this 
```